app/main/views/services.py

```diff
@@ -432,6 +433,7 @@ def update_section_submission(service_id, section_id):
     else:
         update_data.update(uploaded_documents)
 
+    if not errors and has_changes_to_save(section, draft, update_data):
```

Comment:
@TheDoubleK it was always like that. Getting validation errors from non-upload fields requires sending the request to the API, which might be valid and get saved while the documents aren't, leaving the page partially completed. Not showing all of the form errors seems to be a better option in comparison.